src/engine/QueryPlanner.cpp

```diff
@@ -1078,47 +1071,44 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::getOrderByRow(
   const vector<SubtreePlan>& previous = dpTab[dpTab.size() - 1];
   vector<SubtreePlan> added;
   added.reserve(previous.size());
-  for (size_t i = 0; i < previous.size(); ++i) {
+  for (const auto& i : previous) {
     SubtreePlan plan(_qec);
     auto& tree = *plan._qet;
-    plan._idsOfIncludedNodes = previous[i]._idsOfIncludedNodes;
-    plan._idsOfIncludedFilters = previous[i]._idsOfIncludedFilters;
+    plan._idsOfIncludedNodes = i._idsOfIncludedNodes;
+    plan._idsOfIncludedFilters = i._idsOfIncludedFilters;
     if (pq._orderBy.size() == 1 && !pq._orderBy[0]._desc) {
-      size_t col = previous[i]._qet->getVariableColumn(pq._orderBy[0]._key);
-      const std::vector<size_t>& previousSortedOn =
-          previous[i]._qet->resultSortedOn();
-      if (previousSortedOn.size() > 0 && col == previousSortedOn[0]) {
+      size_t col = i._qet->getVariableColumn(pq._orderBy[0]._key);
+      const std::vector<size_t>& previousSortedOn = i._qet->resultSortedOn();
+      if (!previousSortedOn.empty() && col == previousSortedOn[0]) {
         // Already sorted perfectly
-        added.push_back(previous[i]);
+        added.push_back(i);
       } else {
-        auto sort = std::make_shared<Sort>(_qec, previous[i]._qet, col);
-        tree.setVariableColumns(previous[i]._qet->getVariableColumns());
+        auto sort = std::make_shared<Sort>(_qec, i._qet, col);
+        tree.setVariableColumns(i._qet->getVariableColumns());
         tree.setOperation(QueryExecutionTree::SORT, sort);
-        tree.setContextVars(previous[i]._qet->getContextVars());
+        tree.setContextVars(i._qet->getContextVars());
         added.push_back(plan);
       }
     } else {
       vector<pair<size_t, bool>> sortIndices;
       for (auto& ord : pq._orderBy) {
-        sortIndices.emplace_back(pair<size_t, bool>{
-            previous[i]._qet->getVariableColumn(ord._key), ord._desc});
+        sortIndices.emplace_back(
+            pair<size_t, bool>{i._qet->getVariableColumn(ord._key), ord._desc});
```

Comment:
with `emplace_back` you don't need to explicitly write `pair<size_t, ...`, just
`emplace_back(variableColumn(ccc), ord._desc`